### DISCUSSION TITLE
Why not rename ReducerProtocol to Reducer right away?

### DISCUSSION THREAD

**Discussion by chernousov-m** [pointfreeco/swift-composable-architecture#1654] [General]
Since you've already made a few breaking changes ([here](https://github.com/pointfreeco/swift-composable-architecture/commit/b8294b89d594fbd486c449b9f30203461e923145) you changed the generic parameters order, and [here](https://github.com/pointfreeco/swift-composable-architecture/commit/5d687b6b41b5c128f92d80d42171a9767e6dbb80) you added the name `assert` to the parameter in `TestStore`'s send method) I don't see any reason why you should wait with the Reducer renaming
Am I missing something?

**Comment by tgrapperon** [comment]
Hey @chernousov-m! The `Reducer` struct is currently getting in the way and will require you to refactor existing code, often in a non trivial way. The change is planned for v1.0, as you can see in #1477, as this will be the "canonical" the code-breaking release. This will work well with the way SwiftPM automatically resolves its versions, as `from:` is equivalent to "up to next major". For this reason, requiring `from: 0.45.0` will never be upgraded to `1.0.0` without an explicit user action, and users who have not upgraded their code should be able build it without changes even after 1.0 is released.

**Reply by tgrapperon** [reply]
@chernousov-m, looking at the two examples of breaking changes you mentioned, the added `assert` parameters to `TestStore`'s `send` and `receive` have default values, so it shouldn't be a breaking change.
But I have troubles identifying the generic signature that has changed in the very large  `ReducerProtocol` commit. Can you be more specific?

**Reply by chernousov-m** [reply]
The TestStore. Generic parameters are changed

**Reply by chernousov-m** [reply]
Those are breaking nonetheless. Are you familiar with Hyrum’s law? Those who are not ready to refactor yet are able to fix the version of TCA

**Reply by tgrapperon** [reply]
Ah yeah, the `TestStore` has its `State` and `Action` gathered. I guess the change was deemed OK because the vast majority of the uses are using inference and are likely not requiring you to update your code.

**Comment by mbrandonw** [comment]
Hi @chernousov-m, we believe that breaking changes fall on a spectrum, and are not an absolute.

The breaking change of adding the `assert` external argument name only affects people who wrote their store assertions like so:

```swift
await store.send(.tap, {
  $0.count += 1
})
```

We think this is a tiny minority of people, especially since Xcode automatically expands closures to be the trailing syntax.

And as for the re-ordering of generics, I'm not actually able to see that. Both before and after #1599 I'm seeing the same generics:

```swift
<State, Action, ScopedState, ScopedAction, Environment>
```

But, if there really was a change in generics, it would only affect people who decided to explicitly spell out generics when creating test stores, or when extending `TestStore`. Again we feel that is a very tiny minority.

On the other hand, renaming `ReducerProtocol` would affect 100% of people immediately and would be very disruptive to people building their applications. It's just not necessary to introduce that kind of disruption, and we have outlined our plans for renaming in this discussion #1477.
